lib/SIL/TypeLowering.cpp

```diff
@@ -1139,7 +1139,7 @@ namespace {
   /// Build the appropriate TypeLowering subclass for the given type,
   /// which is assumed to already have been lowered.
   class LowerType
-    : public TypeClassifierBase<LowerType, const TypeLowering *>
+    : public TypeClassifierBase<LowerType, TypeLowering *>
```

Comment:
Last time I tried to plumb through something like this I had to change a lot more code. In particular it looks like `class LowerType` has a number of recursive calls to TC.getTypeLowering(), and we're not passing the expansion down through them.

Reply:
Look at https://github.com/apple/swift/pull/5524, where I was trying to do the same thing but with a generic signature.

Reply:
Hm, as I see it, the Expansion is passed down recursively (with classifyType).
Except in visitTupleType - maybe this is a bug?
Anyway, in case it's not passed down in some cases, it just makes the analysis more conservative.